### PR TITLE
Update version detection for subscriptions-core

### LINKED ISSUE #5225: Update API for version detection in subscriptions-core library
### Description

The subscriptions-core library has [updated its API](https://github.com/Automattic/woocommerce-subscriptions-core/issues/239) to add some clarity around versioning. 

`WC_Subscriptions_Core_Plugin::instance()->get_plugin_version()`needs to be changed to `WC_Subscriptions_Core_Plugin::instance()->get_library_version()``

### PR DESCRIPTION
**Pull request by brucealdridge**
Fixes #5225 

#### Changes proposed in this Pull Request

Update version detection in subscriptions-core to use the new API.

Based on a quick check of the code the only place version detection is used is `WC_Payments_Subscriptions_Utilities::get_subscriptions_core_version()` and this doesn't appear to be called. 

This has been updated with a backward compatible check for completeness.

#### Testing instructions

As this code doesn't appear to be used, testing is not really possible.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: includes/compat/subscriptions/trait-wc-payments-subscriptions-utilities.php [hunk 1]
```diff
@@ -120,13 +120,17 @@ public function get_subscriptions_plugin_version() {
 	}
 
 	/**
-	 * Gets the version of the Subscriptions base library running.
-	 *
-	 * This may be the version of the package in WC Payments or WC Subscriptions. Which ever one happens to be loaded.
+	 * Gets the version of the Subscriptions core library running.
 	 *
 	 * @return null|string The core Subscriptions libary version.
```
Comment (Jinksi) [inline includes/compat/subscriptions/trait-wc-payments-subscriptions-utilities.php]:
```suggestion
	 * @return null|string The core Subscriptions library version.
```
Suggestion: fix typo in docblock.

Comment (Jinksi) [reply includes/compat/subscriptions/trait-wc-payments-subscriptions-utilities.php]: Or perhaps this could be `The Subscriptions core library version`.

**Review by Jinksi — APPROVED** [review]
This LGTM 🏆 

FWIW I tested this by running the following in WP Console:
```php
<?php
class TestYo {
    use WC_Payments_Subscriptions_Utilities;
};
$test = new TestYo();
$test->get_subscriptions_core_version();
```

- [x] WCPay activated and not WCS, it returned `'5.1.0'`.
- [x] WCPay activated and WCS 4.7.0 it returned `'5.1.0'`.
- [x] WCPay activated and WCS 4.6.0 (pre-`get_library_version()`) it returned `'4.6.0'`.

File: includes/compat/subscriptions/trait-wc-payments-subscriptions-utilities.php (hunk 1 above)
Comment (haszari) [reply includes/compat/subscriptions/trait-wc-payments-subscriptions-utilities.php]: Maybe even `subscriptions-core` to be 100% consistent naming? Up to y'all.

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5227` or branch name `issue/5225` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=issue/5225) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 986fada5e716327d109ad969af57afbdda37b537
- Build time: 2023-04-24 09:47:42 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Pull request merged**